entgql/internal/todo/generated.go

```diff
@@ -6743,6 +6752,17 @@ func (ec *executionContext) unmarshalInputTodoWhereInput(ctx context.Context, ob
 			if err != nil {
 				return it, err
 			}
+		case "createdToday":
+			var err error
+
+			ctx := graphql.WithPathContext(ctx, graphql.NewPathWithField("createdToday"))
+			data, err := ec.unmarshalOBoolean2ᚖbool(ctx, v)
+			if err != nil {
+				return it, err
+			}
+			if err = ec.resolvers.TodoWhereInput().CreatedToday(ctx, &it, data); err != nil {
+				return it, err
+			}
```

Comment:
@a8m This PR may affect our custom unmarshal task. Because We don't know this field to unmarshal it. https://github.com/ent/contrib/pull/290

Reply:
Agree. Good catch. 

I still think we should not be blocked on this and let @crossworth propose an elegant solution for this.

Reply:
It also requires a call to the field resolver, which thing we don't have access to in the Unmarshal context.

Reply:
I don't fully understand the reason why we should have the custom unmarshal for the complete `WhereFilter`, I got the gist from [this message](https://discord.com/channels/885059418646003782/944956624433467492/960812849016471574). That we need to handle the [Field collection](https://entgo.io/docs/tutorial-todo-gql-field-collection), but from what I understand, the field collection handles what we want to return from the resolver and in Ent's case what we want to eager load, I don't see the reason the filtering would be collected or even know in advance, I think I'm missing something...

With that said, I have a question, does this limitation don't apply to the case when using the generation hooks as well?
If I add a custom field (to the graphql part) using `WithSchemaHook` and override the template to add the required field + the filtering condition, I dont see how Ent is supposed to know the type and how to unmarshal it (of course we could require that every custom type have the matching unmarshal/marshal function to delegate to it).

Alternative ideas for allowing easy custom predicates:
- Improve the template adding similar concepts like [`model/fields/additional`](https://entgo.io/docs/schema-fields/#additional-struct-fields), but for the where struct and where filtering, this way we dont have to override the complete template (I'm not sure if this bypass this limitations).
- Create a new entgql annotation to define this "Custom predicates" to the schema and provide the code generation information about the type, this way we wont need to `extend input TodoWhereInput` but would need to think where to provide the predicate function, maybe in the definition itself (I think this would remove this limitations right?)

But we have to be honest with ourselves, both options above are way more complex and hard to the final user, this PR is a two line deal with very little complexity on the Ent side of things, maybe we should try to solve the other side of the equation.

Reply:
> but from what I understand, the field collection handles what we want to return from the resolver and in Ent's case what we want to eager load, I don't see the reason the filtering would be collected or even know in advance, I think I'm missing something...

Imagine a query like the follows:
```graphql
query {
  users(limit: 100) {
    edges {
      groups(limit: 10, where: {active: true}) {
        totalCount
      }
    }
  }
}
```
Instead of letting gqlgen resolve the groups of each user (N+1 queries), using eager-loading in the fields-collection stage will resolve all groups of all 100 users will be resolved by entgql in a single query. However, the arguments are not "parsed" in this stage ("users" context) and entgql needs to parse them manually (convert from a raw GraphQL representation to a concrete type). You can see how does the generated resolvers are worked in gqlgen generated files.